src/main/java/frc/robot/subsystems/Limelight.java

```diff
@@ -0,0 +1,114 @@
+
+package frc.robot.subsystems;
+
+import java.util.Map;
+
+import edu.wpi.cscore.HttpCamera;
+import edu.wpi.cscore.HttpCamera.HttpCameraKind;
+import edu.wpi.first.networktables.*;
+import edu.wpi.first.wpilibj.shuffleboard.BuiltInLayouts;
+import edu.wpi.first.wpilibj.shuffleboard.ShuffleboardLayout;
+import edu.wpi.first.wpilibj.shuffleboard.ShuffleboardTab;
+import edu.wpi.first.wpilibj2.command.SubsystemBase;
+
+public class Limelight extends SubsystemBase {
+  private final NetworkTable table = NetworkTableInstance.getDefault().getTable("limelight");
+  private final NetworkTableEntry xCoordEntry = table.getEntry("tx");
+  private final NetworkTableEntry yCoordEntry = table.getEntry("ty");
+  private final NetworkTableEntry validEntry = table.getEntry("tv");
+
+  public static final double RANGE_X = 29.8;
+  public static final double RANGE_Y = 24.85;
+
+  public static final double TARGET_DIST = 3;
+  public static final double DIST_ERR = 0;
+  public static final double X_ERR = 0;
+  public static final double Y_ERR = 0;
+
+  private static final double CAMERA_ANGLE = 56.5;
+  private static final double CAMERA_HEIGHT = 23.6;
+  private static final double TARGET_HEIGHT = 84;
```

Comment:
Have these been calibrated?
How variable are they, should they be in MoPrefs?

Reply:
I think RANGE_X and RANGE_Y are constants of the Limelight.  All the others are based on our robot.